tests/onebox.go

```diff
@@ -998,3 +1030,15 @@ func sdkClientFactoryProvider(
 		dynamicconfig.WorkerStickyCacheSize.Get(dc),
 	)
 }
+
+func (c *temporalImpl) overrideDynamicConfigByKey(t *testing.T, name dynamicconfig.Key, value any) {
+	existingValues := c.dcClient.GetValue(name)
+	c.dcClient.OverrideValueByKey(name, value)
+	t.Cleanup(func() {
+		if len(existingValues) > 0 {
+			c.dcClient.OverrideValueByKey(name, existingValues)
```

Comment:
just thinking.. if there was a static override with a plain value, this takes the return value from GetValue, and passes it back to OverrideValueByKey, it'll get wrapped in a []ConstrainedValue, so not identical to how it was before. that's totally fine though, it'll work as expected (as long as Cleanup is called like a stack, which it is).

Reply:
Yeah, it should be fine. We could wrap the value in []ConstrainedValue when it is first inserted so that it stays consistent.